packages/dds/matrix/src/test/utils.ts

```diff
@@ -222,3 +224,25 @@ export function insertFragmented(
 
 	return matrix;
 }
+
+/**
+ * Creates a local matrix with the specified size and initial value.
+ */
+export function createLocalMatrix(
+	id: string,
+	size: number,
+	initialValue: string,
+): ISharedMatrix & IChannel {
+	const matrix = matrixFactory.create(new MockFluidDataStoreRuntime(), id);
+	for (let i = 0; i < size; i++) {
+		matrix.insertRows(i, 1);
+		matrix.insertCols(i, 1);
+	}
```

Comment:
The matrix APIs allow you to insert multiple rows and columns at once. I think you can replace this loop with:
```suggestion
	matrix.insertRows(0, size);
	matrix.insertCols(0, size);
```

Reply:
I tried that. Seems like it still makes test cases out-of-boundary because `matrix.insertRows(0, x)` will update the same index if `this.rows` and `this.cols` properties in matrix. We need this for loop

Reply:
Sorry, I'm not sure I understand. Where specifically do we get an out-of-bounds error? Per the documentation on `insertRows` and `insertCols`, this _should_ work. You just need to make sure the starting index is `0`.